flow_action_components/SendBetterEmail/force-app/main/default/classes/SendBetterEmail.cls

```diff
@@ -70,6 +83,9 @@ public without sharing class SendBetterEmail {
 
     // Collect all inputs for SOQL
     for (SendBetterEmailRequest curRequest : requests) {
+      if (curRequest.sendAsync && !isFuture) {
+        continue;
```

Comment:
no need to query for requests that are not to be processed, if isFuture (being in future context) the request should be processed